api/app/frontend.py

```diff
@@ -66,8 +67,18 @@ class SPAStaticFiles(StaticFiles):
     """
     async def get_response(self, path: str, scope: Scope) -> Response:
         # Call the method on the base class.
-        response = await super().get_response(path, scope)
+        # TODO: This is all terrible. Static HTML should be served up by caddy.
+        try:
+            response = await super().get_response(path, scope)
+        except HTTPException as httpException:
+            # In starlette >= 0.17 and exception is thrown when it's a 404.
+            if httpException.status_code == 404:
+                logger.debug('serving up root for %s', path)
+                request = Request(scope)
+                return await get_index(request)
+            raise
```

Comment:
Does this exception with this change or without, get logged to as a message in RC?

Reply:
Doesn't get logged. We have a ticket #406 to deal with exactly this. I'll move that to the top too.